sequencesender/sequencesender.go

```diff
@@ -96,19 +100,58 @@ func (s *SequenceSender) tryToSendSequence(ctx context.Context) {
 			mTxResultLogger := ethtxmanager.CreateMonitoredTxResultLogger(ethTxManagerOwner, result)
 			mTxResultLogger.Error("failed to send sequence, TODO: review this fatal and define what to do in this case")
 		}
+
+		if len(result.Txs) == 1 {
+			var txL1BlockNumber uint64
+			var txHash common.Hash
+			for _, tx := range result.Txs {
+				txL1BlockNumber = tx.Receipt.BlockNumber.Uint64()
+				txHash = tx.Tx.Hash()
+				break
+			}
+
+			// wait L1 confirmation blocks
+			log.Infof("waiting %d L1 block confirmations for sequence [%d-%d], L1 block: %d, tx: %s",
+				s.cfg.SequenceL1BlockConfirmations, s.lastSequenceInitialBatch, s.lastSequenceEndBatch, txL1BlockNumber, txHash)
+			for {
+				lastL1BlockHeader, err := s.etherman.GetLatestBlockHeader(ctx)
+				if err != nil {
+					log.Errorf("failed to get last L1 block number, err: %v", err)
+				} else {
+					lastL1BlockNumber := lastL1BlockHeader.Number.Uint64()
+
+					if lastL1BlockNumber >= txL1BlockNumber+s.cfg.SequenceL1BlockConfirmations {
+						log.Infof("continuing, last L1 block: %d", lastL1BlockNumber)
+						break
+					}
+				}
+				time.Sleep(waitRetryGetL1Block)
+			}
+
+			lastSCBatchNum, err := s.etherman.GetLatestBatchNumber()
+			if err != nil {
+				log.Warnf("failed to get from the SC last sequenced batch number, err: %v", err)
+				return
+			}
+
+			if lastSCBatchNum != s.lastSequenceEndBatch {
+				s.halt(ctx, fmt.Errorf("last sequenced batch from SC %d doesn't match last sequenced batch sent %d", lastSCBatchNum, s.lastSequenceEndBatch))
+			}
+		} else {
+			s.halt(ctx, fmt.Errorf("monitored txs length is %d and should be <= 1", len(result.Txs)))
+		}
```

Comment:
following up the suggestion above, this can also be changed to:

```go
} else {
    s.halt(ctx, fmt.Errorf("monitored tx %v doesn't have transactions to be checked", result.ID))
}
```